src/vopr.zig

```diff
@@ -713,6 +728,30 @@ pub const Simulator = struct {
                         }
                     }
                 }
+
+                // Potential primaries repair faulty prepares in their journal till op_repair_min
+                // before transitioning to normal status (even if commit_min == commit_max).
+                // Check if replica was stuck repairing such faulty prepares.
+                const op_repair_min_slot = replica.journal.slot_for_op(replica.op_repair_min());
+                const slots = vsr.SlotRange{
+                    .head = op_repair_min_slot,
+                    .tail = replica.journal.slot_with_op(replica.op).?,
+                };
+                // Checked separately as SlotRange.contains doesn't handle empty ranges.
+                const range_empty = slots.head.index == slots.tail.index;
+
+                var iterator = replica.journal.dirty.bits.iterator(.{ .kind = .set });
+                while (iterator.next()) |index| {
+                    const header = replica.journal.header_for_op(index);
+                    const slot_in_repair_range =
+                        (!range_empty and slots.contains(.{ .index = index })) or
+                        (range_empty and index == op_repair_min_slot.index);
+                    if (header != null and slot_in_repair_range) {
+                        if (missing_prepare_op == null or missing_prepare_op.? > header.?.op) {
+                            missing_prepare_op = header.?.op;
+                        }
+                    }
+                }
```

Comment:
> Is it possible that there is a slot in the range with !dirty but with the wrong/missing header? (e.g. an op from a previous wrap.)

Ah yes, it should be only for a potential primary. This code is becoming so branchy though... 😞

> Is it possible that there is a slot in the range with !dirty but with the wrong/missing header? (e.g. an op from a previous wrap.)

Mhmmmm, that is a brilliant point! I had completely missed that, since we're actually repairing all _ops_ from op_repair_min → op, we would actually have to iterate on ops here, and use `header_with_op`. I will fix this.

Reply:
This comment also fixes the same bug in https://github.com/tigerbeetle/tigerbeetle/pull/2414. I've actually been feeling a bit awkward about replica.zig code being copied into test code, but haven't been able to find a better way to do this in vopr.zig. 🤔